packages/admin-ui/src/masked-input/use-formatter.ts

```diff
@@ -0,0 +1,61 @@
+import { useMemo } from 'react'
+
+const MASK_INPUT_SYMBOL = '_'
+
+export interface UseFormatterProps {
+  mask: string
+  accept: RegExp
+}
+
+export function useFormatter(props: UseFormatterProps) {
+  const { mask, accept } = props
+  const formatter = useMemo(
+    () =>
+      createFormatter({
+        mask,
+        accept,
+      }),
+    [mask, accept]
+  )
+
+  return formatter
+}
+
+function createFormatter(props: UseFormatterProps) {
+  const { mask, accept } = props
+
+  function formatter(value: string) {
+    return value
+      .split('')
+      .map((char, i) => {
+        accept.lastIndex = 0
+
+        if (i > mask.length - 1) {
```

Comment:
I think it's better to read, just a suggestion
```suggestion
        if (i >= mask.length) {
```

Reply:
Agreed.